File: app/src/main/java/com/hxw/wanandroid/base/GlobalConfigModule.java
package com.hxw.wanandroid.base;

import android.content.Context;

import com.bumptech.glide.GlideBuilder;
import com.google.gson.GsonBuilder;
import com.hxw.core.base.ConfigModule;
import com.hxw.wanandroid.WanApi;

import org.jetbrains.annotations.NotNull;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;

/**
 * @author hxw on 2018/7/18.
 */
public class GlobalConfigModule implements ConfigModule {
    @Override
    public void configGson(@NotNull Context context, @NotNull GsonBuilder builder) {
        builder.setDateFormat("yyyy-MM-dd HH:mm:ss");
    }

    @Override
    public void configOkHttp(@NotNull Context context, @NotNull OkHttpClient.Builder builder) {

    }

    @Override
    public void configRetrofit(@NotNull Context context, @NotNull Retrofit.Builder builder) {
        builder.baseUrl(WanApi.BASEURL);
    }

    @Override
    public void applyGlideOptions(@NotNull Context context, @NotNull GlideBuilder builder) {

    }
}

File: app/src/main/java/com/hxw/wanandroid/base/GlobalConfigModule.kt
package com.hxw.wanandroid.base

import android.content.Context
import android.content.SharedPreferences
import com.google.gson.GsonBuilder
import com.hxw.core.base.ConfigModule
import com.hxw.wanandroid.WanApi
import com.hxw.wanandroid.base.cookies.CookiesManager
import com.hxw.wanandroid.mvp.host.HostSettingActivity
import okhttp3.OkHttpClient
import org.koin.core.KoinComponent
import org.koin.core.get
import retrofit2.Retrofit
import timber.log.Timber
import java.util.concurrent.TimeUnit

/**
 * @author hxw
 * @date 2018/7/18
 */
class GlobalConfigModule : ConfigModule, KoinComponent {
    override fun configGson(context: Context, builder: GsonBuilder) {
        builder.setDateFormat("yyyy-MM-dd HH:mm:ss")
    }

    override fun configOkHttp(context: Context, builder: OkHttpClient.Builder) {
        builder.cookieJar(CookiesManager(context))
            .connectTimeout(60, TimeUnit.SECONDS)
            .writeTimeout(60, TimeUnit.SECONDS)
            .readTimeout(60, TimeUnit.SECONDS)
    }

    override fun configRetrofit(context: Context, builder: Retrofit.Builder) {
        val sp = get<SharedPreferences>()
        val str = sp.getString(HostSettingActivity.IPUSE, WanApi.BASEURL)
        Timber.d(str)
        builder.baseUrl(WanApi.BASEURL)
    }
}
